src/resdk/collection_tables.py

```diff
@@ -417,27 +433,51 @@ def _expression_file_url(self, data: Data, exp_type: str) -> str:
                 f"Data {data.slug} has multiple expressions of type {exp_type}!"
             )
 
-        return urljoin(self.resolwe.url, f"data/{data.id}/{exp_files[0]}")
+        return f"{data.id}/{exp_files[0]}"
+
+    def _get_exp_urls(self, uris):
+        query_params = "&".join(f"uri={uri}" for uri in uris)
+        url = urljoin(self.resolwe.url, "resolve_uris") + f"?{query_params}"
+        uris_response = self.resolwe.session.get(url, auth=self.resolwe.auth)
+        uris_response.raise_for_status()
+        return json.loads(uris_response.content.decode("utf-8"))
 
-    def _download_expressions(self, exp_type: str) -> pd.DataFrame:
+    async def _download_file(self, url, session, sample_name):
+        async with session.get(url) as response:
+            response.raise_for_status()
+            with BytesIO() as f:
+                f.write(await response.content.read())
+                f.seek(0)
+                df = pd.read_csv(f, sep="\t", compression="gzip")
+                df = df.set_index("Gene").T
+                df.index = [sample_name]
+        return df
+
+    async def _download_expressions(self, exp_type: str) -> pd.DataFrame:
         """Download expression files and marge them into a pandas DataFrame.
 
         :param exp_type: expression type
         :return: table with expression data, genes in columns, samples in rows
         """
         df_list = []
-        for data in tqdm(self._data, desc="Downloading expressions", ncols=100):
-            response = self.resolwe.session.get(
-                self._expression_file_url(data, exp_type), auth=self.resolwe.auth
-            )
-            response.raise_for_status()
-            with BytesIO() as f:
-                f.write(response.content)
-                f.seek(0)
-                df_ = pd.read_csv(f, sep="\t", compression="gzip")
-                df_ = df_.set_index("Gene").T
-                df_.index = [data._original_values["entity"]["name"]]
-                df_list.append(df_)
+        for i in tqdm(range(0, len(self._data), EXP_ASYNC_CHUNK_SIZE)):
```

Comment:
You should add back `tqdm` settings so that the user knows what is being downloaded and so that this progress bar is consistent with the gene mapping progress bar.

Reply:
Good point, fixed.